FrontEnd/src/Components/Profile/StudentSection.jsx

```diff
@@ -0,0 +1,177 @@
+import React, { useState } from 'react';
+import Box from '@mui/material/Box';
+import Modal from '@mui/material/Modal';
+import Button from '@mui/material/Button';
+import Typography from '@mui/material/Typography';
+import AddIcon from '@mui/icons-material/Add';
+import StudentItem from './StudentItem';
+import StudentProfile from '../../Pages/ProfilePage/StudentProfile';
+import EditStudentProfile from '../../Pages/ProfilePage/EditStudentProfile';
+import DeleteDialog from './DeleteDialog';
+import { deleteStudent, findStudents } from '../../api/students';
+import { defaultStudentInfo } from '../../utils/constants';
+
+function StudentSection({
+  user, students, setStudents, setSuccessOpen, setErrorOpen, setInfoOpen, setAlertMessage,
+}) {
+  const [openEditModal, setOpenEditModal] = useState(false);
+  const [currentStudent, setCurrentStudent] = useState(defaultStudentInfo);
+  const [openDeleteDialog, setOpenDeleteDialog] = useState(false);
+  const [isEditing, setIsEditing] = useState(false);
+  const [addStudent, setAddStudent] = useState(false);
+
+  const handleOpenDialog = () => {
+    setOpenDeleteDialog(true);
+  };
+
+  const handleCloseDialog = () => {
+    setOpenDeleteDialog(false);
+  };
```

Comment:
this could be only 1 function

Reply:
Yes, switched to `handleOpenDeleteDialogState`